app/frontend/src/features/messages/surfing/NewHostRequest.tsx

```diff
@@ -30,106 +46,178 @@ const useStyles = makeStyles((theme) => ({
       marginTop: theme.spacing(2),
     },
   },
+  title: {
+    marginBottom: theme.spacing(1),
+  },
+  request: {
+    display: "flex",
+    flexDirection: "row",
+  },
+  date: {
+    marginLeft: theme.spacing(2),
+    marginRight: theme.spacing(2),
+  },
+  requestField: {
+    marginTop: theme.spacing(2),
+  },
+  send: {
+    display: "flex",
+    justifyContent: "flex-end",
+    marginTop: theme.spacing(2),
+  },
 }));
 
-export default function NewHostRequest() {
-  const classes = useStyles();
-
-  const userId = +useParams<{ userId: string }>().userId;
+interface NewHostRequestProps {
+  setIsRequesting: (value: boolean) => void;
+  user: User.AsObject;
+}
 
-  const { data: host, isLoading: hostLoading, error: hostError } = useUser(
-    userId
-  );
-  const title = host
-    ? `Request to be hosted by ${firstName(host.name)}`
-    : undefined;
+export default function NewHostRequest({
+  setIsRequesting,
+  user,
+}: NewHostRequestProps) {
+  const classes = useStyles();
+  const isPostBetaEnabled = process.env.REACT_APP_IS_POST_BETA_ENABLED;
+  const [numVisitors, setNumVisitors] = useState(1);
 
-  const { register, control, handleSubmit, errors: formErrors } = useForm<
-    Required<CreateHostRequestReq.AsObject>
-  >({ defaultValues: { toUserId: userId } });
+  const {
+    control,
+    errors,
+    getValues,
+    handleSubmit,
+    register,
+    setValue,
+    watch,
+  } = useForm<Required<CreateHostRequestReq.AsObject>>({
+    defaultValues: { toUserId: user.userId },
+  });
 
   useEffect(() => register("toUserId"));
 
-  const history = useHistory();
-
-  const mutation = useMutation<
+  const { error, isSuccess, mutate } = useMutation<
     number,
     GrpcError,
     Required<CreateHostRequestReq.AsObject>
-  >(
-    (data: Required<CreateHostRequestReq.AsObject>) =>
-      service.requests.createHostRequest(data),
-    {
-      onSuccess: (hostRequestId) => {
-        history.push(routeToHostRequest(hostRequestId));
-      },
-    }
+  >((data: Required<CreateHostRequestReq.AsObject>) =>
+    service.requests.createHostRequest(data)
   );
 
-  const onSubmit = handleSubmit((data) => mutation.mutate(data));
+  const { isLoading: hostLoading, error: hostError } = useUser(user.userId);
+
+  const onSubmit = handleSubmit((data) => mutate(data));
 
-  const dateValidate = (stringDate: string) =>
-    validateFutureDate(stringDate) || "Must be a valid date in the future.";
+  const guests = Array.from({ length: 8 }, (_, i) => {
+    const num = i + 1;
+    return (
+      <MenuItem key={num} value={num} ref={register}>
+        {num}
+      </MenuItem>
+    );
+  });
+
+  const watchFromDate = watch("fromDate", new Date().toString());
+  useEffect(() => {
```

Comment:
Changing the type back to string means this will need modifying, but you can do `new Date("YYYY-MM-DD")` then do the tomorrow stuff